### DISCUSSION TITLE
Install to different (larger) partition?

### DISCUSSION THREAD

**Discussion by rudolphpienaar** [termux/termux-app#2437] [Ideas]
Some devices have a constrained ``/data`` partition size. I'm thinking here specifically of a google chromecast TV with a tiny 4GB on board storage space. I have added an external 128GB flash drive and formatted this as internal storage (i.e. ``ext4``). It would be amazing if termux could install to this larger (different) internal storage space and then symlink back to ``/data/data/com.termux``.

Is anything like this possible? Thought about?

Best
Rudolph

**Comment by agnostic-apollo** [comment]
Well, normally external sd cards/usbs are not supported since they are formatted as `FAT` and don't have symlink or permission/ownership, but since you have formatted it as `ext4`, you could try moving `$PREFIX` files to `/storage/XXXX-XXXX/Android/data/com.termux` (possibly `~/storage/external-1` for you) after running [`termux-setup-storage`](https://wiki.termux.com/wiki/Termux-setup-storage) and then symlink `/data/data/com.termux/files/usr` to it. There might be some reasons why this would not work, haven't thought much on it.

The other way is using usb to mount an `ext4` image  with termux user ownership at `/data/data/com.termux/files/usr`, but that requires root access. Check the `Storage Requirements` section at end of https://github.com/termux/termux-packages/pull/7227#issuecomment-893022283.

Although, `root` access is likely not possible for you if you have received feb 2021 update, unless some other method is found.
https://www.xda-developers.com/chromecast-with-google-tv-bootloader-unlock-exploit/

Edit: Symlink the directory itself instead of the files under `$PREFIX`.

**Comment by olav2021** [comment]
I am running out of space on my device storage, too.
Maybe it is possible to delete some big files - cache files.

Or should one say, that termux is not suited for devices with less than 10gb internal storage space?

especially python pip is using a lot of space, because it is downloading various versions of packages, before successfully building one. Somehow pip cache purge is not helping much.

Whats the best way of freeing up space on termux?

**Comment by agnostic-apollo** [comment]
If adoptable storage for external sd card is supported on the device, then should be able to use https://www.reddit.com/r/termux/comments/12hty88/comment/jfuv5mg/ without root.
